lighthouse/helpers/plates.py

```diff
@@ -111,12 +117,24 @@ def get_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
 def get_positive_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
     query_filter = copy.deepcopy(POSITIVE_SAMPLES_MONGODB_FILTER)
     query_filter[FIELD_PLATE_BARCODE] = plate_barcode
-
     samples_for_barcode = find_samples(query_filter)
 
     return samples_for_barcode
 
 
+def count_positive_samples(plate_barcode: str) -> int:
+    query_filter = copy.deepcopy(POSITIVE_SAMPLES_MONGODB_FILTER)
+    query_filter[FIELD_PLATE_BARCODE] = plate_barcode
+    samples_for_barcode = count_samples(query_filter)
+
+    return samples_for_barcode
+
+
+def has_sample_data(plate_barcode: str) -> bool:
+    sample_count = count_samples({FIELD_PLATE_BARCODE: plate_barcode})
+    return sample_count > 0
+
+
```

Comment:
These methods are really nice and readable :) I guess if you tried to speed it up by using bulk queries, it would probably sacrifice some of the readability. 

Did you see my question in Slack about whether you'd tried any aggregation queries on UAT? It might be OK actually since you are just passing 96 plate barcodes - I think the issues I had were when querying against larger numbers.

You could also 'bulkify' the positive samples query without using an aggregation, by passing a list of plate barcodes and then looping through the returned data - which I guess would be quicker than doing 96 queries.

Reply:
Yep, I played around a bit with the aggregate query, and got it working using the Result column (See comment above), nice and fast. Could probably import the POSITIVE_SAMPLES_MONGODB_FILTER into the query as well, but didn't look into that. Rich said current performance was acceptable, and wanted to get a version 1 out rather than trying to optimize.

Things will be more robust once the versioning stuff is done and we're persisting the decisions to Mongo, so I'll revisit this then. SHouldn't take too long to make the change.